web/client/components/geostory/contents/EditorCustomizations/CustomEditorLink.jsx

```diff
@@ -0,0 +1,411 @@
+                        <DropdownOption
+                            active={section.id === currentSelectOption}
+                            value={section.id}
+                            key={section.id}
+                        >
+                            {section.title || section.contents[0].title || section.contents[0].id}
+                        </DropdownOption>
+                    ))}
+                </Dropdown>
+            </div>
+        );
+    }
+
+    toggle = () => {
+        this.setState(prev => ({
+            expanded: !prev.expanded
+        }));
+    }
+}
+
+class LayoutComponent extends Component {
+  static propTypes = {
+      expanded: PropTypes.bool,
+      doExpand: PropTypes.func,
+      doCollapse: PropTypes.func,
+      onExpandEvent: PropTypes.func,
+      config: PropTypes.object,
+      onChange: PropTypes.func,
+      currentState: PropTypes.object,
+      translations: PropTypes.object,
+      modalHandler: PropTypes.any
+  };
+
+  state = {
+      showModal: false,
+      linkTarget: '',
+      linkTitle: '',
+      attributes: {
+          type: '',
+          params: ''
+      },
+      linkTargetOption: this.props.config.defaultTargetOption,
+      showExternalLinkOptions: false,
+      currentSelectOption: 'Select link target'
+  };
+
+  componentDidUpdate(prevProps) {
+      if (prevProps.expanded && !this.props.expanded) {
+          this.setState({
+              showModal: false,
+              linkTarget: '',
+              linkTitle: '',
+              attributes: {
+                  type: '',
+                  params: ''
+              },
+              linkTargetOption: this.props.config.defaultTargetOption,
+              showExternalLinkOptions: false,
+              currentSelectOption: 'Select link target'
+          });
+      }
+  }
+
+  removeLink = () => {
+      const { onChange } = this.props;
+      onChange('unlink');
+  };
+
+  addLink = () => {
+      const { onChange } = this.props;
+      const { linkTitle, linkTarget, linkTargetOption, attributes } = this.state;
+      onChange('link', linkTitle, linkTarget, linkTargetOption, attributes);
+  };
+
+  updateValue = event => {
+      this.setState({
+          [`${event.target.name}`]: event.target.value
+      });
+  };
+
+  onSelectionChange = (option) => {
+      // always begin reset the showExternalLinkOptions state
+      this.setState({ showExternalLinkOptions: false });
+      if (option === EXTERNAL_LINK) {
+          this.setState({
+              linkTarget: '',
+              attributes: {
+                  type: '',
+                  params: ''
+              },
+              showExternalLinkOptions: true,
+              currentSelectOption: option });
+      } else {
+          this.setState({
+              linkTarget: option,
+              attributes: {
+                  type: 'scrollTo',
+                  params: `${option}`
+              },
```

Comment:
I think we should use [`data-`](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes) prefix with custom attributes. Could we use directly here the `data-geostory-interaction-type` and `data-geostory-interaction-params`? I'm asking because these attributes are specifically created for the section of geostory.

```js
attributes: {
    'data-geostory-interaction-type': 'scrollTo',
    'data-geostory-interaction-params': `${option}`
},
```